src/librustc_resolve/lib.rs

```diff
@@ -1104,7 +1124,7 @@ impl ModuleOrUniformRoot<'_> {
     fn same_def(lhs: Self, rhs: Self) -> bool {
         match (lhs, rhs) {
             (ModuleOrUniformRoot::Module(lhs),
-             ModuleOrUniformRoot::Module(rhs)) => lhs.def() == rhs.def(),
+             ModuleOrUniformRoot::Module(rhs)) => lhs.def_id() == rhs.def_id(),
```

Comment:
just to confirm, this change is ok, because the `DefKind` would be the same if the `DefId` is the same?

Reply:
Yes, `DefKind`s must be the same for same `DefId`s.

Reply:
Especially since this is a `Module`, so a `mod`/`enum`/`trait`, which don't even have the potential ambiguity that constructors can.